### PR TITLE
Files tab - improve view and findability

### PR DESCRIPTION
**Pull request by sanjaytkbabu**
# Description
PADS-24: Files tab - improve view and findability
Describe your changes in detail:
Added list view to the files tab 
New upload button added
Added button to switch between grid and list views
Added search and sort functionality for files in both views
Changed list documents api to fetch created_by information
Changed created documents api to save information of user to the db
Modularized delete document view and functions

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/views/SubmissionView.vue [hunk 1]
```diff
@@ -36,6 +40,10 @@ const activeTab: Ref<number> = ref(Number(props.initialTab));
 const loading: Ref<boolean> = ref(true);
 const noteModalVisible: Ref<boolean> = ref(false);
 const permitModalVisible: Ref<boolean> = ref(false);
+const gridView = ref(false);
+const searchTag = ref('');
+const sortOrder: Ref<number> = ref(Number(-1));
+const sortType: Ref<string> = ref('createdAt');
```
Comment (wilwong89) [inline frontend/src/views/SubmissionView.vue:46]: Make a local object to reference the different sort types. eg. SORT_CREATED_AT, SORT_USER. Remove magic strings that way.

File: frontend/src/views/SubmissionView.vue [hunk 2]
```diff
@@ -54,9 +62,80 @@ onMounted(async () => {
...
+  return a < b ? 1 : -1;
+}
+
+const filteredDocuments = computed(() => {
+  let tempDocuments = getDocuments.value;
+  tempDocuments = tempDocuments.filter((x) => {
+    return searchTag.value ? x.filename.toLowerCase().includes(searchTag.value.toLowerCase()) : x;
+  });
+  switch (sortType.value) {
+    case 'filename': {
```
Comment (wilwong89) [inline frontend/src/views/SubmissionView.vue:88]: Per prior comment, avoid magic strings in the rest of the switch case.

File: frontend/src/views/SubmissionView.vue [hunk 3]
```diff
@@ -88,17 +167,96 @@ onMounted(async () => {
...
+        </div>
+      </div>
+      <div
+        v-if="gridView"
+        class="grid nested-grid"
+      >
+        <DataTable
+          v-if="gridView"
+          class="remove-padding"
+          sort-field="createdAt"
```
Comment (wilwong89) [inline frontend/src/views/SubmissionView.vue:216]: Per prior comment, remove magic strings. Check other similar props for the same.

File: frontend/src/views/SubmissionView.vue [hunk 4]
```diff
@@ -10,11 +12,13 @@ import PermitCard from '@/components/permit/PermitCard.vue';
...
 import SubmissionForm from '@/components/submission/SubmissionForm.vue';
-import { Button, TabPanel, TabView } from '@/lib/primevue';
+import { Button, TabPanel, TabView, IconField, InputIcon, InputText } from '@/lib/primevue';
 import { submissionService, documentService, noteService, permitService } from '@/services';
 import { useSubmissionStore } from '@/store';
 import { RouteNames } from '@/utils/constants';
-
+import { formatDateLong } from '@/utils/formatters';
+import DeleteDocument from '@/components/file/DeleteDocument.vue';
+import { filesize } from 'filesize';
```
Comment (kyle1morel) [inline frontend/src/views/SubmissionView.vue:21]: Import ordering

File: frontend/src/views/SubmissionView.vue [hunk 5]
```diff
@@ -1,6 +1,8 @@
 <script setup lang="ts">
 import { storeToRefs } from 'pinia';
-import { onMounted, ref } from 'vue';
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
```
Comment (kyle1morel) [inline frontend/src/views/SubmissionView.vue:4]: PrimeVue components should be importable through `@/lib/primevue`

File: frontend/src/views/SubmissionView.vue [hunk 6]
```diff
@@ -54,9 +62,80 @@ onMounted(async () => {
   submissionStore.setNotes(notes);
   submissionStore.setPermits(permits);
   submissionStore.setPermitTypes(permitTypes);
-
   loading.value = false;
 });
+
+// Datatable filter(s)
+
+// State
```
Comment (kyle1morel) [inline frontend/src/views/SubmissionView.vue:70]: Remove unnecessary comments.

File: frontend/src/views/SubmissionView.vue [hunk 7]
```diff
@@ -54,9 +62,80 @@ onMounted(async () => {
...
+      if (sortOrder.value == 1) {
+        tempDocuments = tempDocuments.sort((a, b) => ascendingComparator(a.user.toLowerCase(), b.user.toLowerCase()));
+      } else {
+        tempDocuments = tempDocuments.sort((a, b) => descendingComparator(a.user.toLowerCase(), b.user.toLowerCase()));
+      }
+      break;
+    }
+  }
+  return tempDocuments;
+});
```
Comment (kyle1morel) [inline frontend/src/views/SubmissionView.vue:138]: Structurally it's good. Stylistically, we could cut down on a lot of repeated code by creating one sort function that takes in the sortOrder to determine which way to sort.

**Comment by Subin1Doo** [comment]
Looking great!!
Some minor adjustments please :)

- Center align the delete icon with the word "Action."

- List / grid view icon - make them bigger
- Hover over list/ grid icon-> I think we should add another visual cue here.. light blue circle/ square behind the icon? (I'll add this to Figma)

- The length/ size of the sorting bar changes when you switch views, grid view sort bar is longer
- Grid view sort bar should be the same length as list view sort bar
- Grid view sort bar, can we make it like the way it is on Figma? All "sort by" columns pushed to the left so they are all closer to each other (List view sort bar looks good)

- Grid view - for each file, bottom padding looks greater than top padding (the area around file name, info, and delete icon), can we close the white space by a tiny bit so it's equal to the top 

- Delete confirmation pop-up -> confirm button -> change to red

Thank you!

**Review by kyle1morel — APPROVED** [review]

**Pull request merged**